src/gui/src/components/explorer-grid/selected-item/focused-view/use-focus-state.tsx

```diff
@@ -0,0 +1,22 @@
+import { useEffect, useState } from 'react'
+
+export const useFocusState = (): {
+  focused: boolean
+  setFocused: (focused: boolean) => void
+} => {
+  const [focused, setFocusedState] = useState<boolean>(() => {
+    if (typeof window === 'undefined') return false
```

Comment:
what is the goal / intention of this check here?

Reply:
Server-side rendering

Reply:
in this case I think we should go with a more robust feature detection logic, such as: https://developer.mozilla.org/en-US/docs/Web/API/Web_Storage_API/Using_the_Web_Storage_API#feature-detecting_localstorage

also it's worth noting that we are already using local storage in many plces in the app, so we should pb have an abstraction that makes sure it always passes by feature detection first and then it should be used consistently across the app

I think it's fine to land that refactor later but I'd rather keep consistency and if we have not yet used this feature detection pattern we should not make it as a one-off just here